tensorbay/label/tests/test_catalog.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+
+from ...utility import NameOrderedDict
+from .. import (
+    Box2DSubcatalog,
+    Box3DSubcatalog,
+    Catalog,
+    ClassificationSubcatalog,
+    Keypoints2DSubcatalog,
+    Polygon2DSubcatalog,
+    Polyline2DSubcatalog,
+    SentenceSubcatalog,
+)
+from ..attributes import AttributeInfo
+from ..supports import CategoryInfo, KeypointsInfo
+
+_CATALOG_DATA = {"BOX2D": {}, "BOX3D": {}, "POLYGON2D": {}, "POLYLINE2D": {}, "SENTENCE": {}}
+_CLASSIFICATION_CATALOG_DATA = {
+    "CLASSIFICATION": {
+        "categories": [
+            {
+                "name": "Test",
+            }
+        ]
+    }
+}
+_KEYPOINTS2D_CATALOG_DATA = {
+    "KEYPOINTS2D": {
+        "keypoints": [
+            {
+                "number": 5,
+            }
+        ]
+    }
+}
+
+
+@pytest.fixture
+def categories():
+    categories = NameOrderedDict()
+    categories.append(CategoryInfo("Test"))
+    return categories
+
+
+@pytest.fixture
+def keypoints():
+    return KeypointsInfo(
+        5,
+    )
+
+
+class TestCatalog:
+    def test_bool(self):
+        catalog = Catalog()
+        assert catalog.__bool__() == False
+
+        catalog.box2d = Box2DSubcatalog()
+        assert catalog.__bool__() == True
+
+    def test_loads(self):
+        catalog_1 = Catalog.loads(_CATALOG_DATA)
+        assert catalog_1.box2d.is_tracking == False
+        assert catalog_1.box3d.is_tracking == False
+        assert catalog_1.polygon2d.is_tracking == False
+        assert catalog_1.polyline2d.is_tracking == False
+        assert catalog_1.sentence.is_sample == False
```

Comment:
Use is_instance() to check is the corresponding object build correctly.
Don't forget classification and keypoints2d.